django_irods/views.py

```diff
@@ -354,6 +354,9 @@ def download(request, path, rest_call=False, use_async=True, use_reverse_proxy=T
             return response
 
         if not res.is_federated:
+            # track download count
+            res.update_download_count()
```

Comment:
agh, I loathe this download method.  We keep adding complexity and dependencies to it and we already know it's a mess.  Perhaps this tracking could be moved to the front door of when a download link is hit instead of in here when the file is served.  I'd like to minimize how much django_irods uses hydroshare objects.

Reply:
The concern that I have moving the download tracking inside the initial view function in hs_core is that it would count download even if the download fails with the django_irods. I guess it may not be that critical if the download count is not very accurate. I am OK with moving it out of django_irods.

Reply:
@horsburgh - any thoughts on how accurate you would like the download count to be?  @pkdash is correct that by moving this count to the front door, it would still count failed downloads.

Reply:
@sblack-usu - I guess it should be as accurate as we can make it. A failed download attempt isn't really a download. Do we anticipate a lot of failed download attempts? If this artificially inflates the download count a little, it probably isn't a big deal because a failed download is still someone wanting to download. But, if we anticipate this happening a lot, we should probably try to prevent it.

Reply:
@horsburgh I don't anticipate lot of download failures in general. However, if a resource or file type gets corrupted for some reason, then any attempt to download resource bag, files or folder for that resource would most likely fail until the resource or file type gets fixed. 

@sblack-usu It seems the view function in hs_core that supports file download is not used for resource bag download. For resource bag download the download view function in django_irods is directly used. I guess for now I have to keep the download count in django_irods.

Reply:
ok